tantar/src/main/java/org/opencadc/tantar/policy/RecoverFromStoragePolicy.java

```diff
@@ -0,0 +1,108 @@
+ *  modify it under the terms of         modifier suivant les termes de
+ *  the GNU Affero General Public        la “GNU Affero General Public
+ *  License as published by the          License” telle que publiée
+ *  Free Software Foundation,            par la Free Software Foundation
+ *  either version 3 of the              : soit la version 3 de cette
+ *  License, or (at your option)         licence, soit (à votre gré)
+ *  any later version.                   toute version ultérieure.
+ *
+ *  OpenCADC is distributed in the       OpenCADC est distribué
+ *  hope that it will be useful,         dans l’espoir qu’il vous
+ *  but WITHOUT ANY WARRANTY;            sera utile, mais SANS AUCUNE
+ *  without even the implied             GARANTIE : sans même la garantie
+ *  warranty of MERCHANTABILITY          implicite de COMMERCIALISABILITÉ
+ *  or FITNESS FOR A PARTICULAR          ni d’ADÉQUATION À UN OBJECTIF
+ *  PURPOSE.  See the GNU Affero         PARTICULIER. Consultez la Licence
+ *  General Public License for           Générale Publique GNU Affero
+ *  more details.                        pour plus de détails.
+ *
+ *  You should have received             Vous devriez avoir reçu une
+ *  a copy of the GNU Affero             copie de la Licence Générale
+ *  General Public License along         Publique GNU Affero avec
+ *  with OpenCADC.  If not, see          OpenCADC ; si ce n’est
+ *  <http://www.gnu.org/licenses/>.      pas le cas, consultez :
+ *                                       <http://www.gnu.org/licenses/>.
+ *
+ *
+ ************************************************************************
+ */
+
+package org.opencadc.tantar.policy;
+
+import org.opencadc.inventory.Artifact;
+import org.opencadc.inventory.storage.StorageMetadata;
+import org.opencadc.tantar.Reporter;
+import org.opencadc.tantar.ValidateEventListener;
+
+
+/**
+ * Policy to ensure that a recovery from Storage (in the event of a disaster or a new site is brought online) will
+ * dictate what goes into the Inventory Database.
+ * TODO: Determine if this implementation is required and complete the logic.
+ * TODO: jenkinsd 2020.04.15
+ */
+public class RecoverFromStoragePolicy extends ResolutionPolicy {
+
+    public RecoverFromStoragePolicy(ValidateEventListener validateEventListener, Reporter reporter) {
+        super(validateEventListener, reporter);
+    }
+
+    /**
+     * Use the logic of this Policy to correct a conflict caused by the two given items.  One of the arguments can
+     * be null, but not both.
+     *
+     * @param artifact        The Artifact to use in deciding.
+     * @param storageMetadata The StorageMetadata to use in deciding.
+     * @throws Exception For any unknown error that should be passed up.
+     */
+    @Override
+    public void resolve(final Artifact artifact, final StorageMetadata storageMetadata) throws Exception {
+        if (artifact != null) {
```

Comment:
This simple implementation has to consider the scenario where both artifact and storageMetadata exist (eg it was run, killed/died, and restarted and is redoing some previously recovered entries). So it does need to handle both args not null as a potential replace or update.

Reply:
Thanks.  If both args are not null, it will be assumed that the policy wasn't switched or anything, it just wasn't completed.

Reply:
Full replace is wrong if it is just fixing some mutable metadata. It will ultimately cause other sites to delete files and re-download them when they didn't change.